File: order-service/src/main/java/com/paulopacifico/orderservice/messaging/api/PaymentFailedEvent.java
package com.paulopacifico.orderservice.messaging.api;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

public record PaymentFailedEvent(
        UUID eventId,
        Long orderId,
        String orderNumber,
        BigDecimal amount,
        String reason,
        OffsetDateTime occurredAt
) {
}

File: inventory-service/src/main/kotlin/com/paulopacifico/inventoryservice/messaging/api/PaymentFailedEvent.kt
package com.paulopacifico.inventoryservice.messaging.api

import java.math.BigDecimal
import java.time.OffsetDateTime
import java.util.UUID

data class PaymentFailedEvent(
    val eventId: UUID,
    val orderId: Long,
    val orderNumber: String,
    val amount: BigDecimal,
    val reason: String,
    val occurredAt: OffsetDateTime,
)
